src/main/java/org/opensearch/security/filter/SecurityFilter.java

```diff
@@ -183,6 +183,10 @@ private <Request extends ActionRequest, Response extends ActionResponse> void ap
                 threadContext.putTransient(ConfigConstants.OPENDISTRO_SECURITY_ORIGIN, Origin.LOCAL.toString());
             }
 
+            if (threadContext.getTransient(ConfigConstants.USE_JDK_SERIALIZATION) == null) {
```

Comment:
Can we just set this to a default instead? Why do we ever want it to be nullable?

Reply:
How can we set a default transient in thread context?

Reply:
A lot many `ConfigConstants` are used as keys for header values. I don't think we have default values for these.